doc/releases/changelog-dev.md

```diff
@@ -438,6 +438,9 @@
 * Simplify method for `Exp` now returns an operator with the correct number of Trotter steps, i.e. equal to the one from the pre-simplified operator.
   [(#5831)](https://github.com/PennyLaneAI/pennylane/pull/5831)
 
+* Fix bug where `CompositeOp.overlapping_ops` sometimes puts overlapping ops in different groups, leading to incorrect results returned by `LinearCombination.eigvals()`
```

Comment:
Are there any other consequences of this bug, or did it only affect eigvals?

Reply:
It's only used in `eigvals` and `matrix`, and the original bug report was `eigvals` returning incorrect results.

Reply:
If it also fixes a bug occurring in `matrix`, I think it would be good to mention in the changelog.

Reply:
Actually the only time `overlapping_ops` is used in `matrix` is in `Prod`, and after careful inspection, the special implementation of `overlapping_ops` in `Prod` actually seemed to be correct, so no bug there.

Reply:
Brilliant, thank you!